app/models/event_stream.rb

```diff
@@ -59,11 +61,19 @@ def self.event_groups
 
   def self.group_and_level(event_type)
     level = :detail # the level is detail as default
-    group, _ = event_groups.find do |_k, value|
-      GROUP_LEVELS.detect { |lvl| value[lvl]&.include?(event_type) }.tap do |level_found|
-        level = level_found if level_found
-      end
-    end
+    egroups = event_groups
+
+    group = egroups.find do |_, value|
+      GROUP_LEVELS
+        .detect { |lvl, _| value[lvl]&.include?(event_type) }
+        .tap { |level_found, _| level = level_found || level }
+    end&.first
+
+    group ||= egroups.find do |_, value|
+      GROUP_LEVELS
+        .detect { |_, re_lvl| value[re_lvl]&.find { |regex| Regexp.new(regex).match(event_type) } }
```

Comment:
So if regex is not given, we end up with `Regexp.new(nil)` which raises TypeError, right?

Reply:
If my ruby serves me, then no. If `value[re_lvl]` returns `nil`, because regex was not given, then `find` is not executed at all.